### PR TITLE
[#9917] Loading bar persists when courses of the student are soft-deleted

### LINKED ISSUE #9917: Loading bar persists when courses of the student are soft-deleted
**Environment**: 
`master` branch at commit https://github.com/TEAMMATES/teammates/pull/9897/commits/59c356e84ccfbf5863797fb5ca80270be506ff41.

**Steps to reproduce**
Create a new course, enroll a student, and move the course to the recycle bin. Then login as the student and navigate to the student's home page.

**Expected behaviour**
Display a message showing no courses available for the student.

**Actual behaviour**
The top progress bar persists and there is no error message.
image

### PR DESCRIPTION
**Pull request by aishaadambek**
Fixes #9917

**Outline of Solution**

1. Added a check whether the `courses` array (returned from the `/courses` endpoint) is empty;
2. If the `courses` array is indeed empty,

- hid the loading bar;
- display an explanatory message to a student.

### CODE REVIEW

**Comment by aishaadambek** [comment]
Hi team!

As specified in the [Development Workflow document](https://github.com/TEAMMATES/teammates/blob/master/docs/process.md#step-5-following-up), if a PR is not assigned to a reviewer in 24 hours, a contributor can leave a comment to remind about the open PR. I also requested a review from the reviewers suggested on the right.

Looking forward to working on this!

File: src/web/app/pages-student/student-home-page/__snapshots__/student-home-page.component.spec.ts.snap [hunk 1]
```diff
@@ -389,7 +389,18 @@ exports[`StudentHomePageComponent should snap with default fields 1`] = `
   studentFeedbackSessionStatusSubmitted={[Function String]}
   timezoneService={[Function TimezoneService]}
 >
+  <div>
+     No courses you are enrolled in. Please refer to the
```
Comment (madanalogy) [inline src/web/app/pages-student/student-home-page/__snapshots__/student-home-page.component.spec.ts.snap]:
```suggestion
     You are not enrolled in any courses. Please refer to the 
```

File: src/web/app/pages-student/student-home-page/__snapshots__/student-home-page.component.spec.ts.snap [hunk 2]
```diff
@@ -389,7 +389,18 @@ exports[`StudentHomePageComponent should snap with default fields 1`] = `
...
+  <div>
+     No courses you are enrolled in. Please refer to the 
+    <a
+      href="/web/student/help"
+      ng-reflect-router-link="/web/student/help"
+      routerlink="/web/student/help"
+    >
+      Help page
+    </a>
+     if you have any questions.
```
Comment (madanalogy) [inline src/web/app/pages-student/student-home-page/__snapshots__/student-home-page.component.spec.ts.snap]:
```suggestion
     if you need more information.
```

File: src/web/app/pages-student/student-home-page/student-home-page.component.html [hunk 3]
```diff
@@ -1,85 +1,91 @@
...
-      <span class="float-sm-right">
-        <a routerLink="../course" [queryParams]="{courseid: course.courseId}"
-           class="btn btn-light btn-sm button" ngbTooltip='View and edit information regarding your team'>
-          View team
-        </a>
-      </span>
-    </div>
+<div *ngIf="courses.length == 0; else coursesContainer">
+  No courses you are enrolled in. Please refer to the <a routerLink="/web/student/help">Help page</a>
+  if you have any questions.
```
Comment (madanalogy) [inline src/web/app/pages-student/student-home-page/student-home-page.component.html]: Similarly for this paragraph as well

File: src/web/app/pages-student/student-home-page/__snapshots__/student-home-page.component.spec.ts.snap [hunk 4]
```diff
@@ -389,7 +389,18 @@ exports[`StudentHomePageComponent should snap with default fields 1`] = `
...
   timezoneService={[Function TimezoneService]}
 >
+  <div>
+     No courses you are enrolled in. Please refer to the 
+    <a
+      href="/web/student/help"
+      ng-reflect-router-link="/web/student/help"
+      routerlink="/web/student/help"
+    >
+      Help page
```
Comment (madanalogy) [inline src/web/app/pages-student/student-home-page/__snapshots__/student-home-page.component.spec.ts.snap]: Any particular reason for capitalising the "H" in "Help"?

File: src/web/app/pages-student/student-home-page/__snapshots__/student-home-page.component.spec.ts.snap (hunk 1 above)
Comment (aishaadambek) [reply src/web/app/pages-student/student-home-page/__snapshots__/student-home-page.component.spec.ts.snap]: Done.

File: src/web/app/pages-student/student-home-page/__snapshots__/student-home-page.component.spec.ts.snap (hunk 2 above)
Comment (aishaadambek) [reply src/web/app/pages-student/student-home-page/__snapshots__/student-home-page.component.spec.ts.snap]: Done.

File: src/web/app/pages-student/student-home-page/student-home-page.component.html (hunk 3 above)
Comment (aishaadambek) [reply src/web/app/pages-student/student-home-page/student-home-page.component.html]: Yes, no problems :) Done.

File: src/web/app/pages-student/student-home-page/__snapshots__/student-home-page.component.spec.ts.snap (hunk 4 above)
Comment (aishaadambek) [reply src/web/app/pages-student/student-home-page/__snapshots__/student-home-page.component.spec.ts.snap]: Done.

File: src/web/app/pages-student/student-home-page/student-home-page.component.ts [hunk 5]
```diff
@@ -76,31 +76,35 @@ export class StudentHomePageComponent implements OnInit {
       entitytype: 'student',
     };
     this.httpRequestService.get('/courses', paramMap).subscribe((resp: Courses) => {
-      for (const course of resp.courses) {
-        this.feedbackSessionsService.getFeedbackSessionsForStudent(course.courseId)
-            .pipe(finalize(() => this.loadingBarService.hideLoadingBar()))
-            .subscribe((fss: FeedbackSessions) => {
-              const sortedFss: FeedbackSession[] = this.sortFeedbackSessions(fss);
+      if (resp.courses.length === 0) {
```
Comment (rrtheonlyone) [inline src/web/app/pages-student/student-home-page/student-home-page.component.ts]:
```
if (resp.courses.length === 0) {
        this.loadingBarService.hideLoadingBar();
        return;
}
//rest of course 
```

Comment (aishaadambek) [reply src/web/app/pages-student/student-home-page/student-home-page.component.ts]: Oops, done!

**Comment by aishaadambek** [comment]
Changes made, conflicts resolved. PR is ready for review again.

**Comment by aishaadambek** [comment]
> Really nice work with this PR!! Thank you for you contribution :)
> 
> I can help give some preliminary comments before passing onto a more senior developer. Code-wise I only have one minor nit (shown above).
> 
> You also have to fix some conflicts here - you can also use `git rebase` to clean up your commit history.

Thanks for having a look at this!
Changes made, conflicts resolved. PR should be ready for the review now :)

**Comment by rrtheonlyone** [comment]
Requesting final review from @wkurniawan07

**Pull request merged**